src/framework/components/rigid-body/system.js

```diff
@@ -547,7 +583,12 @@ class RigidBodyComponentSystem extends ComponentSystem {
             const numHits = collisionObjs.size();
             for (let i = 0; i < numHits; i++) {
                 const body = Ammo.castObject(collisionObjs.at(i), Ammo.btRigidBody);
-                if (body) {
+
+                if (body && body.entity) {
+                    if (options.filterTags && !body.entity.has(...options.filterTags) || options.filterCallback && !options.filterCallback(body.entity)) {
+                        continue;
+                    }
```

Comment:
Wondering whether you meant?
```suggestion
                if (body) {
                    if (body.entity) {
                        if (options.filterTags && !body.entity.has(...options.filterTags) || options.filterCallback && !options.filterCallback(body.entity)) {
                            continue;
                        }
                    }
```

Reply:
No, in fact the current behaviour to return a result even if there are no entity linked does not make sense as the body that was hit is never returned, therefore the user will have information about a hit present but no information about what was actually hit.